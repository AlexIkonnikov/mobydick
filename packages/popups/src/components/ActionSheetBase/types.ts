import {ViewStyle} from 'react-native';
import {TypographyProp} from '@npm/mobydick-typography';
import {ReactElement} from 'react';

export interface IPropsContents {
  title: string;
  textFont?: TypographyProp;
  leftIcon?: ReactElement;
  onPress?(): void;
  radio?: string;
  checkboxList?: string[];
}

export enum IItemType {
  firstItem = 'firstItem',
  innerItem = 'innerItem',
  lastItem = 'lastItem',
  singleItem = 'singleItem',
  cancelItem = 'cancelItem',
}

export interface IPropsItem extends IPropsContents {
  itemType: IItemType;
  style?: ViewStyle;
  disabled?: boolean;
}
