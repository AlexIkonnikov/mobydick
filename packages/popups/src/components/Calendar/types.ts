export interface IChangeDate {
  dateStart: string;
  dateEnd: string;
}
