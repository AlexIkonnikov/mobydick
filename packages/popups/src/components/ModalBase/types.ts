export enum IAlertTypes {
  warning = 'warning',
  check = 'check',
}
