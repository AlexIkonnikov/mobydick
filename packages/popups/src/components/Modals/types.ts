export interface IModalProps {
  title: string;
  descriptionText: string;
  buttonText?: string;
}
