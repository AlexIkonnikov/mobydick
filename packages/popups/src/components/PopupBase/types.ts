import {TextStyle} from 'react-native';
import {TypographyProp} from '@npm/mobydick-typography';

import {IPopup} from '../../types';

export interface ITitlePopup {
  title: string;
  titleStyles?: TextStyle | TextStyle[];
  titleFont?: TypographyProp;
}

export interface IDescriptionTextPopup {
  descriptionText: string;
  descriptionStyles?: TextStyle | TextStyle[];
  descriptionFont?: TypographyProp;
}

export interface IContentTextProps {
  title?: string;
  titleStyles?: TextStyle | TextStyle[];
  titleFont?: TypographyProp;

  descriptionText?: string;
  descriptionStyles?: TextStyle | TextStyle[];
  descriptionFont?: TypographyProp;
}

export interface IPopupCloseIcon {
  onPress(): void;
}

export type IPopupProps = Pick<IPopup, 'overlayStyle'> & {
  onClose: () => void;
};
