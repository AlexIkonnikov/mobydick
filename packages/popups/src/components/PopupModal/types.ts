import {ViewStyle} from 'react-native';
import {ReactNode} from 'react';

export interface IPopupModal {
  style?: ViewStyle;
  isExitIcon?: boolean;
  title?: string;
  titleStyles?: ViewStyle;
  children?: ReactNode;
}
