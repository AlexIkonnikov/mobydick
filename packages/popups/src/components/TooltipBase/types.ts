import {ColorValue, ViewStyle} from 'react-native';

export enum IPosition {
  top = 'top',
  bottom = 'bottom',
}

export enum IPlacement {
  start = 'start',
  center = 'center',
  end = 'end',
}

export interface IArrowViewPopup {
  placement: IPlacement;
  position: IPosition;
  arrowViewStyles?: ViewStyle;
  colorTooltip: ColorValue;
}
