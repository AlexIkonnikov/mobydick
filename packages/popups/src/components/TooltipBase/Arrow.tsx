import React, {FC} from 'react';
import {useStyles} from '@npm/mobydick-styles';
import {View} from '@npm/mobydick-core';
import {ViewStyle} from 'react-native';

import stylesCreate from './stylesCreate';
import {IArrowViewPopup, IPlacement} from './types';

const placementArrow = (placement: IPlacement): ViewStyle => {
  switch (placement) {
    case IPlacement.start:
      return {
        left: 10,
      };
    case IPlacement.end:
      return {
        right: 10,
      };
    case IPlacement.center:
    default:
      return {
        alignSelf: 'center',
      };
  }
};

const Arrow: FC<IArrowViewPopup> = props => {
  const {arrowViewStyles, placement, position} = props;
  const [styles] = useStyles(stylesCreate, position);
  return (
    <View
      style={[
        styles.arrow,
        arrowViewStyles,
        styles.positionStyle,
        placementArrow(placement),
      ]}
    />
  );
};

export default Arrow;
