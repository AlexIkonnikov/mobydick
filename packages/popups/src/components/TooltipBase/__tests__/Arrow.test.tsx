import {render} from '@testing-library/react-native';
import React from 'react';

import Arrow from '../Arrow';
import {IPlacement, IPosition} from '../types';

describe('@npm/mobydick-popups/TooltipBase/Arrow', () => {
  it('should renders correctly top start', () => {
    const {toJSON} = render(
      <Arrow
        placement={IPlacement.start}
        position={IPosition.top}
        arrowViewStyles={{width: 20}}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly bottom center', () => {
    const {toJSON} = render(
      <Arrow
        placement={IPlacement.center}
        position={IPosition.bottom}
        arrowViewStyles={{width: 20}}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly bottom end', () => {
    const {toJSON} = render(
      <Arrow placement={IPlacement.end} position={IPosition.bottom} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
