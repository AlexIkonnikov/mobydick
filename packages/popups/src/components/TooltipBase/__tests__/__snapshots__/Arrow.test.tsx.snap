// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-popups/TooltipBase/Arrow should renders correctly bottom center 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderStyle": "solid",
        "height": 0,
        "position": "absolute",
        "width": 0,
      },
      Object {
        "width": 20,
      },
      Object {
        "borderBottomColor": "transparent",
        "borderBottomWidth": 0,
        "borderLeftColor": "transparent",
        "borderLeftWidth": 12,
        "borderRightColor": "transparent",
        "borderRightWidth": 12,
        "borderTopColor": "#FFF",
        "borderTopWidth": 12,
        "bottom": -12,
      },
      Object {
        "alignSelf": "center",
      },
    ]
  }
/>
`;

exports[`@npm/mobydick-popups/TooltipBase/Arrow should renders correctly bottom end 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderStyle": "solid",
        "height": 0,
        "position": "absolute",
        "width": 0,
      },
      undefined,
      Object {
        "borderBottomColor": "transparent",
        "borderBottomWidth": 0,
        "borderLeftColor": "transparent",
        "borderLeftWidth": 12,
        "borderRightColor": "transparent",
        "borderRightWidth": 12,
        "borderTopColor": "#FFF",
        "borderTopWidth": 12,
        "bottom": -12,
      },
      Object {
        "right": 10,
      },
    ]
  }
/>
`;

exports[`@npm/mobydick-popups/TooltipBase/Arrow should renders correctly top start 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderStyle": "solid",
        "height": 0,
        "position": "absolute",
        "width": 0,
      },
      Object {
        "width": 20,
      },
      Object {
        "borderBottomColor": "#FFF",
        "borderBottomWidth": 12,
        "borderLeftColor": "transparent",
        "borderLeftWidth": 12,
        "borderRightColor": "transparent",
        "borderRightWidth": 12,
        "borderTopColor": "transparent",
        "borderTopWidth": 0,
        "top": -12,
      },
      Object {
        "left": 10,
      },
    ]
  }
/>
`;
